File: 1813-maximum-erasure-value/maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        vector<bool> unique(10001, false);
        int sub_sum = 0;
        int answer = 0;

        auto left = nums.begin();
        auto right = nums.begin();

        while (left != nums.end()) {
            while (right != nums.end() && !unique[*right]) {
                sub_sum += *right;
                unique[*right] = true;
                ++right;
            }

            answer = max(answer, sub_sum);
            unique[*left] = false;
            sub_sum -= *left;
            ++left;
        }

        return answer;
    }
};

File: 1813-maximum-erasure-value/maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        seen = set()
        answer = sub_sum = left = 0

        for right in range(len(nums)):
            while nums[right] in seen:
                seen.remove(nums[left])
                sub_sum -= nums[left]
                left += 1
            seen.add(nums[right])
            sub_sum += nums[right]
            answer = max(answer, sub_sum)

        return answer